pkg/signup/service/signup_service_test.go

```diff
@@ -217,23 +220,33 @@ func (s *TestSignupServiceSuite) TestUserSignupWithInvalidSubjectPrefix() {
 	val := userSignups.Items[0]
 
 	// Confirm that the UserSignup.Name value has been prefixed correctly
-	crc32q := crc32.MakeTable(0xEDB88320)
-	expected := fmt.Sprintf("%x%s", crc32.Checksum([]byte(subject), crc32q), subject)
+	expected := fmt.Sprintf("%x%s", crc32.Checksum([]byte(subject), crc32.IEEETable), subject)
 	require.Equal(s.T(), expected, val.Name)
 	require.False(s.T(), strings.HasPrefix(val.Name, "-"))
 }
-*/
 
-/*
 func (s *TestSignupServiceSuite) TestEncodeUserID() {
 	s.Run("test valid user ID unchanged", func() {
 		userID := "abcde-12345"
-
 		encoded := service.EncodeUserID(userID)
-
 		require.Equal(s.T(), userID, encoded)
 	})
-}*/
+	s.Run("test user ID with invalid characters", func() {
+		userID := "abcde\\*-12345"
+		encoded := service.EncodeUserID(userID)
+		require.Equal(s.T(), "c0177ca4-abcde-12345", encoded)
+	})
+	s.Run("test user ID with invalid prefix", func() {
+		userID := "-1234567"
+		encoded := service.EncodeUserID(userID)
+		require.Equal(s.T(), "ca3e1e0f-1234567", encoded)
+	})
+	s.Run("test user ID that exceeds max length", func() {
+		userID := "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-01234567890123456789"
+		encoded := service.EncodeUserID(userID)
+		require.Equal(s.T(), "e3632025-0123456789abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqr", encoded)
+	})
```

Comment:
Let's also add a test for abc:xyz user id format. This format is used by RH sso.

Reply:
Done